src/pipdeptree/_models/package.py

```diff
@@ -190,14 +231,18 @@ class ReqPackage(Package):
     UNKNOWN_VERSION = "?"
 
     def __init__(self, obj: Requirement, dist: DistPackage | None = None) -> None:
-        super().__init__(obj)
+        if hasattr(obj, "key"):
+            super().__init__(obj.key)
+        else:
+            super().__init__(obj.name)
```

Comment:
Follow what I said [here](https://github.com/tox-dev/pipdeptree/pull/333/files#r1536871750).

Reply:
When I am working on testcases fixture, I meet a condusing problem which I eventually figured it out.
`Mock` object can not be set `name` attr and always return `<Mock name='bar.name' id='140667609844304'>`.
In this case, `MagicMock` need to used.